src/py42/_internal/archive_access.py

```diff
@@ -167,48 +168,107 @@ def _get_children(self, node_id=None):
             show_deleted=True,
         )
 
-    def _get_file_size_info(self, file_ids):
-        file_sizes = []
+
+def _get_default_file_size():
+    return {u"numFiles": 1, u"numDirs": 1, u"size": 1}
+
+
+class _RestorePoller(object):
+    JOB_POLLING_INTERVAL_SECONDS = 1
+
+    def __init__(self, storage_archive_client, device_guid, job_polling_interval=None):
+        self._storage_archive_client = storage_archive_client
+        self._device_guid = device_guid
+        self._job_polling_interval = (
+            job_polling_interval or self.JOB_POLLING_INTERVAL_SECONDS
+        )
+
+
+class FileSizePoller(_RestorePoller):
+    JOB_POLLING_TIMEOUT = 10
+
+    def __init__(
+        self, storage_archive_client, device_guid, job_polling_interval=None,
+    ):
+        super(FileSizePoller, self).__init__(
+            storage_archive_client, device_guid, job_polling_interval
+        )
+
+    def get_file_sizes(self, file_ids, timeout=None):
+        if timeout is None:
+            timeout = self.JOB_POLLING_TIMEOUT
+
+        # Allows a timeout of 0 to ignore file size calculation altogether
+        if not timeout:
+            return None
+
+        sizes = []
+        t0 = time.time()
         for file_id in file_ids:
-            size_data = self._storage_archive_client.get_file_size(
-                self._device_guid, file_id
+            job_id = self._start_poll(file_id)
+            response = self._wait_for_job(job_id)
+            sizes.append(
```

Comment:
Can we kick off all the jobs at once before waiting for them so they can run in parallel?